packages/instantsearch-ui-components/src/components/Carousel.tsx

```diff
@@ -0,0 +1,222 @@
+   */
+  list: string | string[];
+  /**
+   * Class names to apply to each item element
+   */
+  item: string | string[];
+  /**
+   * Class names to apply to both navigation elements
+   */
+  navigation: string | string[];
+  /**
+   * Class names to apply to the next navigation element
+   */
+  navigationNext: string | string[];
+  /**
+   * Class names to apply to the previous navigation element
+   */
+  navigationPrevious: string | string[];
+};
+
+export type CarouselTranslations = {
+  /**
+   * The label of the next navigation element
+   */
+  nextButtonLabel: string;
+  /**
+   * The title of the next navigation element
+   */
+  nextButtonTitle: string;
+  /**
+   * The label of the previous navigation element
+   */
+  previousButtonLabel: string;
+  /**
+   * The title of the previous navigation element
+   */
+  previousButtonTitle: string;
+  /**
+   * The label of the carousel
+   */
+  listLabel: string;
+};
+
+let lastCarouselId = 0;
+
+function generateCarouselId() {
+  return `ais-Carousel-${lastCarouselId++}`;
+}
+
+export function createCarouselComponent({ createElement, useRef }: Renderer) {
+  // This prevents the SVG buttons references from being hoisted
+  // during the build, where `createElement` is not available.
+  const buttonPathCommonProps: JSX.IntrinsicElements['path'] = {
+    fillRule: 'evenodd',
+    clipRule: 'evenodd',
+    fill: 'currentColor',
+  };
+
+  return function Carousel<TObject extends RecordWithObjectID>(
+    props: CarouselProps<TObject>
+  ) {
+    const listRef = useRef?.<HTMLOListElement>(null);
+    const nextButtonRef = useRef?.<HTMLButtonElement>(null);
+    const previousButtonRef = useRef?.<HTMLButtonElement>(null);
+    const carouselIdRef = useRef?.(generateCarouselId());
+
+    const translations: Required<CarouselTranslations> = {
+      listLabel: 'Items',
+      nextButtonLabel: 'Next',
+      nextButtonTitle: 'Next',
+      previousButtonLabel: 'Previous',
+      previousButtonTitle: 'Previous',
+      ...props.translations,
+    };
+    const classNames = props.classNames ?? {};
+
+    function scrollLeft() {
+      if (listRef?.current) {
+        listRef.current.scrollLeft -= listRef.current.offsetWidth * 0.75;
+      }
+    }
+
+    function scrollRight() {
+      if (listRef?.current) {
+        listRef.current.scrollLeft += listRef.current.offsetWidth * 0.75;
+      }
+    }
+
+    function updateNavigationButtonsProps() {
+      if (
+        !listRef?.current ||
+        !previousButtonRef?.current ||
+        !nextButtonRef?.current
+      ) {
+        return;
+      }
+
+      previousButtonRef.current.hidden = listRef.current.scrollLeft <= 0;
+      nextButtonRef.current.hidden =
+        listRef.current.scrollLeft + listRef.current.clientWidth >=
+        listRef.current.scrollWidth;
+    }
+
+    if (props.items.length === 0) {
+      return null;
+    }
+
+    return (
+      <div className={cx('ais-Carousel', classNames.root)}>
```

Comment:
📝 the root class was previously named `uic-HorizontalSlider-container`, I removed the last part to align with what we have in InstantSearch